### PR TITLE
ci: migrate elasticsearch suite to GitLab

### PR DESCRIPTION
**Pull request by brettlangdon**
The major difference between this iteration and the previous is we are properly setting up the elasticsearch and opensearch services.

GitLab doesn't really like when two services are using the same port. It'll be a toss up as to which service you'll end up connecting to. The flakiness before was when we were trying to testing the elasticsearch clients with the opensearch server. Even connecting via the proper alias names didn't improve it.

```
$ echo "http://${TEST_ELASTICSEARCH_HOST}:${TEST_ELASTICSEARCH_PORT}/"
http://elasticsearch:9200/
$ curl "http://${TEST_ELASTICSEARCH_HOST}:${TEST_ELASTICSEARCH_PORT}/" || true
{
  "name" : "runner-rrydb6mn-project-358-concurrent-3-7ux5b2ke",
  "cluster_name" : "docker-cluster",
  "cluster_uuid" : "Gbb1nvm3SoSUraCNAicMPQ",
  "version" : {
    "distribution" : "opensearch",
    "number" : "1.3.13",
    "build_type" : "tar",
    "build_hash" : "07ebac12b8942b51d9d4ada93a95aee76dc1e8e1",
    "build_date" : "2023-09-19T22:10:41.733049Z",
    "build_snapshot" : false,
    "lucene_version" : "8.10.1",
    "minimum_wire_compatibility_version" : "6.8.0",
    "minimum_index_compatibility_version" : "6.0.0-beta1"
  },
  "tagline" : "The OpenSearch Project: https://opensearch.org/"
}
$ echo "http://${TEST_OPENSEARCH_HOST}:${TEST_OPENSEARCH_PORT}/"
http://opensearch:9200/
$ curl "http://${TEST_OPENSEARCH_HOST}:${TEST_OPENSEARCH_PORT}/" || true
{
  "name" : "runner-rrydb6mn-project-358-concurrent-3-7ux5b2ke",
  "cluster_name" : "docker-cluster",
  "cluster_uuid" : "Gbb1nvm3SoSUraCNAicMPQ",
  "version" : {
    "distribution" : "opensearch",
    "number" : "1.3.13",
    "build_type" : "tar",
    "build_hash" : "07ebac12b8942b51d9d4ada93a95aee76dc1e8e1",
    "build_date" : "2023-09-19T22:10:41.733049Z",
    "build_snapshot" : false,
    "lucene_version" : "8.10.1",
    "minimum_wire_compatibility_version" : "6.8.0",
    "minimum_index_compatibility_version" : "6.0.0-beta1"
  },
  "tagline" : "The OpenSearch Project: https://opensearch.org/"
}
```

## Checklist
- [x] PR author has checked that all the criteria below are met
- The PR description includes an overview of the change
- The PR description articulates the motivation for the change
- The change includes tests OR the PR description describes a testing strategy
- The PR description notes risks associated with the change, if any
- Newly-added code is easy to change
- The change follows the [library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html)
- The change includes or references documentation updates if necessary
- Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))

## Reviewer Checklist
- [x] Reviewer has checked that all the criteria below are met 
- Title is accurate
- All changes are related to the pull request's stated goal
- Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes
- Testing strategy adequately addresses listed risks
- Newly-added code is easy to change
- Release note makes sense to a user of the library
- If necessary, author has acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment
- Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)

### CODE REVIEW

**Review by taegyunkim — APPROVED** [review]

File: .gitlab/tests/contrib.yml [hunk 1]
```diff
@@ -3,6 +3,14 @@ algoliasearch:
   variables:
     SUITE_NAME: "algoliasearch"
 
+aredis:
```
Comment (emmettbutler) [inline .gitlab/tests/contrib.yml:6]: is this intentional?

File: .gitlab/tests/contrib.yml [hunk 2]
```diff
@@ -130,6 +85,17 @@ jinja2:
   variables:
     SUITE_NAME: "jinja2"
 
+kafka:
```
Comment (emmettbutler) [inline .gitlab/tests/contrib.yml:88]: This file has a bunch of unrelated changes in it

**Review by emmettbutler — CHANGES_REQUESTED** [review]

File: .gitlab/tests/contrib.yml (hunk 2 above)
Comment (brettlangdon) [reply .gitlab/tests/contrib.yml:88]: I'll revert the alphabetical changes.

**Review by emmettbutler — APPROVED** [review]
